src/course/handlers/get-course-batches-handler.ts

```diff
@@ -1,19 +1,14 @@
 import {ApiRequestHandler, ApiService, HttpRequestType, Request} from '../../api';
 import {Batch, CourseBatchesRequest, CourseServiceConfig} from '..';
 import {CourseBatchesResponse} from '../def/course-batches-response';
-import {ServerProfile} from '../../profile/def/server-profile';
-import {ProfileService} from '../../profile';
-import {AuthService, OAuthSession} from '../../auth';
-import {Observable, of} from 'rxjs';
-import {map, mergeMap, switchMap} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
 
 export class GetCourseBatchesHandler implements ApiRequestHandler<CourseBatchesRequest, Batch[]> {
-    private readonly GET_COURSE_BATCHES = '/batch/list';
```

Comment:
Rather than add the queryParam in the path, possibly use .withParameters({ creatorDetails: '' }) in the requestBuilders. This can prevent it from being overridden later.